### PR TITLE
[KV] Add key concepts section and reference architectures to KV docs

### PR DESCRIPTION
**Pull request by thomasgauvin**

### CODE REVIEW

**Review by kodster28 — APPROVED** [review]

File: content/kv/demos.md [hunk 1]
```diff
@@ -18,7 +18,4 @@ Explore the following {{<glossary-tooltip term_id="demo application">}}demo appl
 
 Explore the following {{<glossary-tooltip term_id="reference architecture">}}reference architectures{{</glossary-tooltip>}} that use KV:
 
-* [A/B-testing using Workers](/reference-architecture/diagrams/serverless/a-b-testing-using-workers/): Use KV to build an A/B testing architecture, where KV is used to store the routing information to the proper origin service based on the incoming request.
```
Comment (kodster28) [inline content/kv/demos.md]: @thomasgauvin, with this change, you do lose the KV-specific descriptions... but figure that gets outweighed by these getting pulled in automatically.

Let me know if that's not the case and happy to flip it back.

Comment (thomasgauvin) [reply content/kv/demos.md]: Thanks, this is more maintainable long term and follows the demos pattern 👍

**Comment by thomasgauvin** [comment]
Will wait on feedback from https://github.com/cloudflare/cloudflare-docs/pull/15911 before merging

**Comment by kodster28** [comment]
Merged in #15911, so think we should be good on this one as well.

File: content/kv/demos.md [hunk 2]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 5
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (thomasgauvin) [inline content/kv/demos.md]:
```suggestion
# Demo applications and reference architectures
```
Following @marciocloudflare recommendations on #15911

**Comment by thomasgauvin** [comment]
@kodster28 just had one last nit based on #15911 feedback, will wait for build then merge

**Comment by thomasgauvin** [comment]
Fixed merge conflict @kodster28

**Pull request merged**